fix: Avoid race condition on filename validation

doubleface opened the pull request:
When pressing enter key on filename creation we would get an unwanted notification for empty filename even if not empty.

I simplified the handling of these events and now :
- we save the file when pressing enter and the file name is not empty
- we display the "no empty file name" notification when pressing enter with an empty file name
- we save the file when bluring the component and the file name is not empty
- we just blur without notification on blur with empty file name

https://www.notion.so/linagora/Toast-when-creating-folder-29562718bad18055a7ccfcf0c523489b
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Bug Fixes**
  * Improved filename input behavior: Enter, Escape and blur now consistently submit or abort based on the field value (whitespace is treated as content). Added a submission guard to prevent duplicate submissions and race conditions; submitting state is reliably cleared on completion or error.

* **Tests**
  * Added comprehensive tests covering Enter, Escape, blur, typing, whitespace edge cases and race-condition scenarios.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Commit:
fix: Avoid race condition on filename validation

When pressing enter key on filename creation we would get an unwanted notification for empty filename even if not empty.

I simplified the handling of these events and now :
- we save the file when pressing enter and the file name is not empty
- we display the "no empty file name" notification when pressing enter with an empty file name
- we save the file when bluring the component and the file name is not empty
- we just blur without notification on blur with empty file name

rezk2ll commented on src/modules/filelist/FilenameInput.jsx:
```diff
@@ -35,11 +34,13 @@ class FilenameInput extends Component {
-    if (e.keyCode === ENTER_KEY && !valueIsEmpty(value)) {
-      this.setState({ hasBeenSubmitedOrAborted: true })
-      this.submit()
+    if (e.keyCode === ENTER_KEY) {
+      if (valueIsEmpty(value)) {
+        this.abort(true)
+      } else {
+        this.submit()
+      }
```
this won't work since React batches the state updates and executes the event handlers an then update the state.

in the case of handleBlur the hasSubmittedOrAborted will always have the previous value and not the value updated by the handleKeyDown event handler

rezk2ll replied:
how about useRef ? :eyes:

rezk2ll replied:
we will do that in a different PR :eyes:

Coussecousse commented on src/modules/filelist/FilenameInput.spec.jsx:
```diff
@@ -0,0 +1,162 @@
+      expect(onAbort).toHaveBeenCalledTimes(1)
+    })
+
+    it('should handle blur correctly without race condition', () => {
+      const { onSubmit, onAbort } = setup()
+      const input = screen.getByRole('textbox')
+
+      // Type some text and blur
+      fireEvent.change(input, { target: { value: 'valid-file' } })
+      fireEvent.blur(input)
+
+      // Should submit without any race condition issues
+      expect(onSubmit).toHaveBeenCalledWith('valid-file')
+      expect(onAbort).not.toHaveBeenCalled()
+    })
+  })
+
+  describe('edge cases', () => {
+    it('should handle empty string value correctly', () => {
+      const { onAbort } = setup({ name: '' })
+      const input = screen.getByRole('textbox')
+
+      // Press Enter with empty string
+      fireEvent.keyDown(input, { keyCode: 13 })
+
+      expect(onAbort).toHaveBeenCalledWith(true)
+    })
+
+    it('should handle whitespace-only value as non-empty', () => {
+      const { onSubmit } = setup()
+      const input = screen.getByRole('textbox')
+
+      // Type whitespace and press Enter
+      fireEvent.change(input, { target: { value: '   ' } })
+      fireEvent.keyDown(input, { keyCode: 13 })
+
+      // Whitespace is considered non-empty by the component
+      expect(onSubmit).toHaveBeenCalledWith('   ')
+    })
+  })
```
write me a fiction about this pr

Commit:
fix: Improve abort behavior in FilenameInput component

Refine the abort logic in the FilenameInput component to differentiate between folder creation and file renaming. Now, when the input is blurred:
- For folder creation with no initial name, it exits without notification.
- For file renaming with an initial name, it shows a notification if the filename is empty.
